pkg/liqo-controller-manager/namespaceMap-controller/namespacemap_controller.go

```diff
@@ -0,0 +1,95 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespacemapctrl
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+)
+
+// NamespaceMapReconciler creates remote namespaces and updates NamespaceMaps Status.
+type NamespaceMapReconciler struct {
+	client.Client
+	Scheme         *runtime.Scheme
+	Mapper         meta.RESTMapper
+	RemoteClients  map[string]client.Client
+	LocalClusterID string
+}
+
+// cluster-role
+// +kubebuilder:rbac:groups=discovery.liqo.io,resources=foreignclusters,verbs=get;list
+// not necessary but it must have role to create/delete remote Namespaces
+// +kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;create;delete
```

Comment:
Not clear to me the meaning of the comment. Additionally, it seems to me the permissions on namespaces are required in the remote clusters, not the local one (hence, they do not need to be added to this role)

Reply:
yes is not clear the comment means exactly what you have said, that marker is only a reminder, that we have to add permission in the remote clusters. I will remove that.